src/addons/addonmessage.cpp

```diff
@@ -101,12 +103,18 @@ void AddonMessage::updateMessageState(State newState) {
 
   QMetaEnum stateMetaEnum = QMetaEnum::fromType<State>();
   QString newStateSetting = stateMetaEnum.valueToKey(newState);
-
+  QString oldStateSetting = stateMetaEnum.valueToKey(m_state);
   m_state = newState;
   emit stateChanged(m_state);
 
   SettingsHolder* settingsHolder = SettingsHolder::instance();
   settingsHolder->setAddonSetting(StateQuery(id()), newStateSetting);
+  emit MozillaVPN::instance()->recordGleanEventWithExtraKeys(
+      GleanSample::addonMessageStateChanged, {{"message_id", id()},
+                                              {"old_state", oldStateSetting},
```

Comment:
Why here you care about the old state and in the addon.cpp you don't?
I think we need to rename State to MessageState to avoid confusion

Reply:
Updated! I was sanity checking things in the Glean debug view but we don't need to care about the old_state.